src/daemon/command_parser_executor.cpp

```diff
@@ -187,17 +187,32 @@ bool t_command_parser_executor::print_block(const std::vector<std::string>& args
 
 bool t_command_parser_executor::print_transaction(const std::vector<std::string>& args)
 {
-  if (args.empty())
+  bool include_hex = false;
+  bool include_json = false;
+  bool unexpected_optional_argument = false;
+
+  for (unsigned int i = 1; i < args.size(); ++i) {
+    if (args[i] == "--with-hex")
+      include_hex = true;
+    else if (args[i] == "--with-json")
+      include_json = true;
+    else
+    {
+      unexpected_optional_argument = true;
+      break;
```

Comment:
printing "bad arg: $arg[i]" here makes it more user friendly and also makes the flow simpler.

Reply:
ok